docs/standard/linq/concepts-terminology-functional-transformation.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Concepts and terminology (functional transformation) - LINQ to XML
+description: Learn the concepts and terminology of pure functional transformations.
+ms.date: 07/20/2015
+ms.assetid: 03defb3a-7e17-4ab1-8efa-4dd66621e860
+---
+
+# Concepts and terminology (functional transformation) (LINQ to XML)
+
+This article introduces the concepts and terminology of pure functional transformations. The functional transformation approach to transforming data yields code that's often quicker to program, more expressive, and easier to debug and maintain than more traditional, imperative programming.
+
+Note that the articles in this section aren't intended to fully explain functional programming. Instead, these articles identify some of the functional programming capabilities that make it easier to transform XML from one shape to another.
+
+## What is pure functional transformation
+
+In *pure functional transformation*, a set of functions, called *pure functions*, define how to transform a set of structured data from its original form into another form. The word "pure" indicates that the functions are *composable*, which requires that they're:
+
+- *Self-contained*, so that they can be freely ordered and rearranged without entanglement or interdependencies with the rest of the program. Pure transformations have no knowledge of or effect upon their environment. That is, the functions used in the transformation have no *side effects*.
+- *Stateless*, so that executing the same function or specific set of functions on the same input will always result in the same output. Pure transformations have no memory of their prior use.
+
+> [!IMPORTANT]
+> In the rest of this tutorial, the term "pure function" is used in a general sense to indicate a programming approach, and not a specific language feature.
+>
+> Note that pure functions must be implemented as methods in C#, and as functions in Visual Basic.
+>
+> You should not confuse pure functions with pure virtual methods in C++. The latter indicates that the containing class is abstract and that no method body is supplied.
+
+### Functional programming
+
+*Functional programming* is a programming approach that directly supports pure functional transformation.
+
+Historically, general-purpose functional programming languages, such as ML, Scheme, Haskell, and F#, have been primarily of interest to the academic community. Although it has always been possible to write pure functional transformations in C# and Visual Basic, the difficulty of doing so has not made it an attractive option to most programmers. In recent versions of these languages, however, new language constructs such as lambda expressions and type inference make functional programming much easier and more productive.
+
+For more information about functional programming, see [Functional programming vs. imperative programming](functional-programming-vs-imperative-programming.md).
```

Comment:
Let's shorten this file to avoid dupe words

```suggestion
For more information about functional programming, see [Functional programming vs. imperative programming](functional-vs-imperative-programming.md).
```